tools/make/packaging.mk

```diff
@@ -0,0 +1,230 @@
+	$(PACKAGING_VARS) AGENTCTL_BINARY=$@ $(MAKE) -f $(PARENT_MAKEFILE) agentctl
+
+#
+# DEB and RPM packages.
+#
+
+ENVIRONMENT_FILE_rpm := /etc/sysconfig/grafana-agent
+ENVIRONMENT_FILE_deb := /etc/default/grafana-agent
+
+# generate_fpm(deb|rpm, package arch, agent arch, output file)
+define generate_fpm =
+	fpm -s dir -v $(PACKAGE_VERSION) -a $(2) \
+		-n grafana-agent --iteration $(PACKAGE_RELEASE) -f \
+		--log error \
+		--license "Apache 2.0" \
+		--vendor "Grafana Labs" \
+		--url "https://github.com/grafana/agent" \
+		-t $(1) \
+		--after-install packaging/$(1)/control/postinst \
+		--before-remove packaging/$(1)/control/prerm \
+		--config-files /etc/grafana-agent.yaml \
+		--config-files $(ENVIRONMENT_FILE_$(1)) \
+		--package $(4) \
+			dist/agent-linux-$(3)=/usr/bin/grafana-agent \
+			dist/agentctl-linux-$(3)=/usr/bin/grafana-agentctl \
+			packaging/grafana-agent.yaml=/etc/grafana-agent.yaml \
+			packaging/environment-file=$(ENVIRONMENT_FILE_$(1)) \
+			packaging/$(1)/grafana-agent.service=/usr/lib/systemd/system/grafana-agent.service
+endef
+
+PACKAGE_VERSION := $(patsubst v%,%,$(VERSION))
+PACKAGE_RELEASE := 1
+PACKAGE_PREFIX  := dist/grafana-agent-$(PACKAGE_VERSION)-$(PACKAGE_RELEASE)
+
+.PHONY: dist-packages
+dist-packages: dist-packages-amd64 \
+               dist-packages-arm64 \
+               dist-packages-armv6 \
+               dist-packages-armv7 \
+               dist-packages-arm64 \
+               dist-packages-ppc64le
+
+.PHONY: dist-packages-amd64
+dist-packages-amd64: dist/agent-linux-amd64 dist/agentctl-linux-amd64
+ifeq ($(USE_CONTAINER),1)
+	$(RERUN_IN_CONTAINER)
+else
+	$(call generate_fpm,deb,amd64,amd64,$(PACKAGE_PREFIX).amd64.deb)
+	$(call generate_fpm,rpm,x86_64,amd64,$(PACKAGE_PREFIX).amd64.rpm)
+endif
+
+.PHONY: dist-packages-arm64
+dist-packages-arm64: dist/agent-linux-arm64 dist/agentctl-linux-arm64
+ifeq ($(USE_CONTAINER),1)
+	$(RERUN_IN_CONTAINER)
+else
+	$(call generate_fpm,deb,arm64,arm64,$(PACKAGE_PREFIX).arm64.deb)
+	$(call generate_fpm,rpm,aarch64,arm64,$(PACKAGE_PREFIX).arm64.rpm)
+endif
+
+# There's no RPM for armv6 so only debs are produced.
+.PHONY: dist-packages-armv6
+dist-packages-armv6: dist/agent-linux-armv6 dist/agentctl-linux-armv6
+ifeq ($(USE_CONTAINER),1)
+	$(RERUN_IN_CONTAINER)
+else
+	$(call generate_fpm,deb,armhf,armv6,$(PACKAGE_PREFIX).armv6.deb)
+endif
+
+.PHONY: dist-packages-armv7
+dist-packages-armv7: dist/agent-linux-armv7 dist/agentctl-linux-armv7
+ifeq ($(USE_CONTAINER),1)
+	$(RERUN_IN_CONTAINER)
+else
+	$(call generate_fpm,deb,armhf,armv7,$(PACKAGE_PREFIX).armv7.deb)
+	$(call generate_fpm,rpm,armhfp,armv7,$(PACKAGE_PREFIX).armv7.rpm)
+endif
+
+.PHONY: dist-packages-ppc64le
+dist-packages-ppc64le: dist/agent-linux-ppc64le dist/agentctl-linux-ppc64le
+ifeq ($(USE_CONTAINER),1)
+	$(RERUN_IN_CONTAINER)
+else
+	$(call generate_fpm,deb,ppc64el,ppc64le,$(PACKAGE_PREFIX).ppc64el.deb)
+	$(call generate_fpm,rpm,ppc64le,ppc64le,$(PACKAGE_PREFIX).ppc64le.rpm)
```

Comment:
it's annoying if fedora and debian use different nomenclature to reference the same architecture, e.g. ppc64le vs ppc64el.  I can't tell if these lines are correct at. glance but it seems it should be double checked.

Reply:
They're correct, see our existing Makefile: 

https://github.com/grafana/agent/blob/cbbfabb8b7844d702409dc62dcc32fd330973ec5/Makefile#L474-L475

https://github.com/grafana/agent/blob/cbbfabb8b7844d702409dc62dcc32fd330973ec5/Makefile#L483-L484

(I triple-checked what I wrote here against the current Makefile, since I definitely agree it's confusing)